### PR TITLE
my requests tab complete

### PR DESCRIPTION
**Pull request by 1ntEgr8**

### CODE REVIEW

File: client/src/components/inventory/HardwareItem.tsx [hunk 1]
```diff
@@ -109,7 +142,24 @@ class HardwareItemBase extends React.Component<Item & HardwareItem, HardwareItem
...
 
+    addCancelledItem(qtyToAdd: number) {
+        this.setState({
+            qtyRemaining: this.state.qtyRemaining + qtyToAdd
+        })
+    }
+
+    componentDidUpdate(prevProps: Item & HardwareItem, prevState: HardwareItemState) {
+        if (this.props.qtyUpdate) {
+            console.log("i am inside componentWillupdate")
```
Comment (evan10s) [inline client/src/components/inventory/HardwareItem.tsx]: Remove console.log statements pls

File: client/src/components/inventory/HardwareItem.tsx [hunk 2]
```diff
@@ -13,10 +13,34 @@ export interface Item {
...
+    LOST = "red",
+    DAMAGED = "red"
+}
+
 interface HardwareItem {
     qtyRemaining: number, // # of this item remaining in our stock
     requestsEnabled: boolean, // whether hardware requests can be made at this time
     toastManager: any, // for making toast notifications
+    addItem: (item: RequestedItem) => void,
+    qtyUpdate: RequestedItem | null,
```
Comment (evan10s) [inline client/src/components/inventory/HardwareItem.tsx]: 1. Took me a while to figure out what's going on with this (updating item quantity after an item request is cancelled), so please add a comment clarifying the purpose
2. For future reference, a way to selectively refresh the quantity remaining for items (or just websockets) would be a better way to do this.  But I appreciate that you took the time to think of this in the first place

File: client/src/components/inventory/HardwareItem.tsx (hunk 1 above)
Comment (1ntEgr8) [reply client/src/components/inventory/HardwareItem.tsx]: lol forgot to do that

File: client/src/components/requests/RequestsList.tsx [hunk 3]
```diff
@@ -1,17 +1,61 @@
...
+import {Header, Icon, Segment, Card, Image, Button, Label, Grid, Divider, Container, Step} from "semantic-ui-react";
+import {RequestedItem, ItemStatus} from "../inventory/HardwareItem"
+
+class RequestsList extends Component<{requestedItemsList: RequestedItem[], removeItem: (index: number) => void}>{
+    getItemStatusString(status: ItemStatus) {
+        switch(status) {
+            case ItemStatus.SUBMITTED: return "Submitted"
+            case ItemStatus.APPROVED: return "Approved"
+            case ItemStatus.DECLINED: return "Declined"
+            case ItemStatus.ABANDONED: return "Abandoned"
```
Comment (evan10s) [inline client/src/components/requests/RequestsList.tsx]:
```suggestion
            case ItemStatus.CANCELLED: return "Cancelled"
```

**Pull request merged**